programs/system/src/lib.rs

```diff
@@ -96,11 +98,10 @@ pub fn entrypoint(
                 program_id,
             } => create_system_account(keyed_accounts, lamports, space, &program_id).map_err(|e| {
                 match e {
-                    SystemError::AccountAlreadyInUse => ProgramError::InvalidArgument,
                     SystemError::ResultWithNegativeLamports => {
                         ProgramError::ResultWithNegativeLamports
                     }
-                    SystemError::SourceNotSystemAccount => ProgramError::InvalidArgument,
+                    e => ProgramError::CustomError(custom_error!(e)),
```

Comment:
I'd prefer this was just:

```rust
e => ProgramError::CustomError(serialize(&e).unwrap()),
```

And that the length check be enforced in runtime.rs. Also, runtime.rs should have a unit-test for that check and resulting truncation.